### PR TITLE
BWS-110: + space delete endpoint

### PR DESCRIPTION
**Pull request by Donymak**
I'm not sure about the return type of delete method in the space service.

### CODE REVIEW

File: backend/src/api/bs/bs.api.ts [hunk 1]
```diff
@@ -63,6 +64,21 @@ const initBsApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
       return rep.send(spaces).status(HttpCode.OK);
     },
   });
+
+  fastify.route({
+    method: HttpMethod.DELETE,
+    url: BSApiPath.SPACES,
+    async handler(req: FastifyRequest<{ Body: BsSpaceDeleteRequestDto }>, rep) {
```
Comment (what1s1ove) [inline backend/src/api/bs/bs.api.ts]: deleting should be by `:id` in params

File: backend/src/data/repositories/space/space.repository.ts [hunk 2]
```diff
@@ -31,6 +31,10 @@ class Space {
     return Space.modelToEntity(created);
   }
 
+  async delete({ name }: { name: string }): Promise<number> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/space/space.repository.ts]:
```suggestion
  async delete(id: string): Promise<number> {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/bs/bs.api.ts [hunk 3]
```diff
@@ -63,6 +64,25 @@ const initBsApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
+
+  fastify.route({
+    method: HttpMethod.DELETE,
+    url: BSApiPath.SPACES,
+    async handler(
+      req: FastifyRequest<{ Querystring: BSSpaceDeleteParamsDto }>,
+      rep,
+    ) {
+      const [, token] = req.headers?.authorization?.split(' ') ?? [];
+      const id = req.query.id;
```
Comment (what1s1ove) [inline backend/src/api/bs/bs.api.ts]:
```suggestion
      const { id } = req.params;
```
should be param

File: backend/src/data/repositories/space/space.repository.ts [hunk 4]
```diff
@@ -31,13 +31,27 @@ class Space {
     return Space.modelToEntity(created);
   }
 
+  async delete({ id }: { id: string }): Promise<number> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/space/space.repository.ts]:
```suggestion
  async delete(id: string): Promise<number> {
```

File: backend/src/data/repositories/space/space.repository.ts [hunk 5]
```diff
@@ -31,13 +31,27 @@ class Space {
...
+    return this.#SpaceModel.query().where({ id }).del();
+  }
+
+  async getSpaceById({ id }: { id: string }): Promise<SpaceEntity> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/space/space.repository.ts]:
```suggestion
  async getSpaceById(id: string): Promise<SpaceEntity> {
```

File: backend/src/services/aws/s3/s3.service.ts [hunk 6]
```diff
@@ -29,6 +31,14 @@ class S3 {
   }): Promise<CreateBucketOutput> {
     return this.#s3Client.send(new CreateBucketCommand({ Bucket: name }));
   }
+
+  public async deleteBucket({
+    name,
+  }: {
+    name: string;
+  }): Promise<DeleteBucketCommandOutput> {
```
Comment (what1s1ove) [inline backend/src/services/aws/s3/s3.service.ts]:
```suggestion
  public async deleteBucket(name: string): Promise<DeleteBucketCommandOutput> {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/bs/bs.api.ts [hunk 7]
```diff
@@ -63,6 +64,25 @@ const initBsApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
...
+    ) {
+      const [, token] = req.headers?.authorization?.split(' ') ?? [];
+      const { id } = req.params;
+
+      await spaceService.delete({
+        id,
+        token,
+      });
+
+      return rep.send({ id: id }).status(HttpCode.OK);
```
Comment (what1s1ove) [inline backend/src/api/bs/bs.api.ts]:
```suggestion
      return rep.send(true).status(HttpCode.OK);
```

File: shared/src/common/enums/api/bs-api-path.enum.ts [hunk 8]
```diff
@@ -1,6 +1,7 @@
 enum BSApiPath {
   ROOT = '/',
   SPACES = '/spaces',
+  SPACE = '/spaces/:id',
```
Comment (what1s1ove) [inline shared/src/common/enums/api/bs-api-path.enum.ts]:
```suggestion
  SPACES_$ID = '/spaces/:id',
```
`$` for dynamic parameters

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

**Pull request closed without merging**